pyglet/graphics/vertexbuffer.py

```diff
@@ -309,5 +309,7 @@ def invalidate(self):
         until the next time the buffer is used, for efficiency).
         """
         buffer = self.buffer
-        buffer._dirty_min = min(buffer._dirty_min, self.start)
-        buffer._dirty_max = max(buffer._dirty_max, self.end)
+        if self.start < buffer._dirty_min:
```

Comment:
Explaining why we aren't using the more concise syntax will save everyone time by preventing reversion attempts.
```suggestion
        # Intentionally avoid min and max functions to increase performance as of 3.8 - 3.12
        if self.start < buffer._dirty_min:
 ```